### PR TITLE
MISC-239 - Initial Development

### PR DESCRIPTION
**Pull request by mattdrees**
This adds the initial Cru checkstyle rules, as well as the derived rules for Wsapi.

For https://jira.cru.org/browse/MISC-239.

### CODE REVIEW

**Comment by mattdrees** [comment]
@wrandall22 @frett I converted this into a PR. That's what I should have done in the first place.

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# ignore maven target directory
+target/
```
Comment (wrandall22) [inline .gitignore:2]: What about IntelliJ files and things like `.DS_STORE`?

**Review by wrandall22 — COMMENTED** [review]
One comment

File: .gitignore (hunk 1 above)
Comment (mattdrees) [reply .gitignore:2]: Yeah, good idea.

I have a global gitignore that ignores those, but probably worth putting them in the project gitignore.

**Review by wrandall22 — APPROVED** [review]

**Review by frett — APPROVED** [review]
Looks good to me

**Comment by mattdrees** [comment]
thanks guys!

**Pull request merged**